altair/vegalite/v5/api.py

```diff
@@ -1998,13 +2030,13 @@ def transform_filter(
             returns chart to allow for chaining
         """
         if isinstance(filter, Parameter):
-            new_filter: dict[str, bool | str] = {"param": filter.name}
+            new_filter: dict[str, Any] = {"param": filter.name}
```

Comment:
Why does this have to be an Any type? Can FilterTransform actually accept anything other than `bool` or `str`?

Reply:
See https://github.com/vega/altair/pull/3480/files/419a4e944f231026322e2c4f137e7a3eb94735e8#r1679846774

I think those errors were self-imposed?

Reply:
Resolved in https://github.com/vega/altair/pull/3480#discussion_r1681450713